modules/camera/web/face_detection.js

```diff
@@ -0,0 +1,306 @@
+                    const mouth = landmarks.getMouth();
+                    const jawOutline = landmarks.getJawOutline();
+                    if (leftEye.length === 0 || rightEye.length === 0 ||
+                        nose.length === 0 || mouth.length === 0 ||
+                        jawOutline.length === 0) {
+                        qualityIssues.push('Some essential facial features are not visible');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 2. Face Within Frame: Ensure the detected face is not too close to the video edges
+                    const margin = 0.05;
+                    const isCompletelyInFrame =
+                        normalizedX >= margin &&
+                        normalizedY >= margin &&
+                        (normalizedX + normalizedWidth) <= (1 - margin) &&
+                        (normalizedY + normalizedHeight) <= (1 - margin);
+                    if (!isCompletelyInFrame) {
+                        qualityIssues.push('Move face away from edges');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 3. Face Alignment (Roll): Check tilt angle (should be less than 6°)
+                    const leftEyeCenter = {
+                        x: leftEye.reduce((sum, p) => sum + p.x, 0) / leftEye.length,
+                        y: leftEye.reduce((sum, p) => sum + p.y, 0) / leftEye.length
+                    };
+                    const rightEyeCenter = {
+                        x: rightEye.reduce((sum, p) => sum + p.x, 0) / rightEye.length,
+                        y: rightEye.reduce((sum, p) => sum + p.y, 0) / rightEye.length
+                    };
+                    const dx = rightEyeCenter.x - leftEyeCenter.x;
+                    const dy = rightEyeCenter.y - leftEyeCenter.y;
+                    const tiltAngle = Math.abs(Math.atan2(dy, dx) * (180 / Math.PI));
+                    if (tiltAngle > 6) {
+                        qualityIssues.push(`Keep face straight (tilt: ${tiltAngle.toFixed(1)}°)`);
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 4. Face Centering (Horizontal): Check if the nose tip is near the center horizontally
+                    const faceCenterX = box.x + (box.width / 2);
+                    const noseTip = nose[Math.floor(nose.length / 2)]; // Use central nose point
+                    const centeredThreshold = box.width * 0.08; // 8% of face width tolerance
+                    if (Math.abs(noseTip.x - faceCenterX) > centeredThreshold) {
+                        qualityIssues.push('Center face horizontally');
+                    } else {
+                        passedChecks++;
+                    }
+
+                    // 5. Eye Openness: Use Eye Aspect Ratio (EAR)
+                    const getEAR = (eye) => {
```

Comment:
please have these as optional parameters for the web for app developers with default values. Do not hardcode like this then the next customer would want something else